### PR TITLE
feat(rpc): added scheme in bls generate rpc + aligned changes for bls fromsecret rpc

### PR DESCRIPTION
**Pull request by ogabrielides**
## What was done?

- `bls generate` and `bls fromsecret` rpcs will return `scheme` used to serialise the public key. Valid returned values are `legacy` and`basic`.
- `bls generate` and `bls fromsecret`rpcs accept an incoming optional boolean argument `legacy` that enforces the use of legacy BLS scheme for the serialisation of the reply even if v19 is active.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by ogabrielides** [comment]
> Looks good. We could maybe "synchronize" it with `bls fromsecret` though i.e.
> 
> 1. accept `legacy` as an optional param
> 2. either return `legacy` instead of `scheme` here or tweak `bls fromsecret` to return `scheme` instead of `legacy` too
> 
> Thoughts?

@UdjinM6 
1. Agreed
2. I don't see `bls fromsecret` returning legacy in the code.. I will add `scheme` in the reply in order to be aligned with `bls generate`

**Comment by ogabrielides** [comment]
@UdjinM6 Done

**Comment by UdjinM6** [comment]
> > Looks good. We could maybe "synchronize" it with `bls fromsecret` though i.e.
> > 
> > 1. accept `legacy` as an optional param
> > 2. either return `legacy` instead of `scheme` here or tweak `bls fromsecret` to return `scheme` instead of `legacy` too
> > 
> > Thoughts?
> 
> @UdjinM6
> 
> 1. Agreed
> 2. I don't see `bls fromsecret` returning legacy in the code.. I will add `scheme` in the reply in order to be aligned with `bls generate`

2. That's because I was looking at my local code 🙈 😄

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/rpc/evo.cpp [hunk 1]
```diff
@@ -1327,9 +1330,15 @@ static UniValue bls_generate(const JSONRPCRequest& request)
...
     bool bls_legacy_scheme = !llmq::utils::IsV19Active(::ChainActive().Tip());
+    if (!request.params[1].isNull()) {
+        RPCTypeCheckArgument(request.params[1], UniValue::VBOOL);
+        bls_legacy_scheme = request.params[1].get_bool();
+    }
     UniValue ret(UniValue::VOBJ);
     ret.pushKV("secret", sk.ToString());
     ret.pushKV("public", sk.GetPublicKey().ToString(bls_legacy_scheme));
+    std::string bls_scheme_str = bls_legacy_scheme ? "legacy" : "basic";
+    ret.pushKV("scheme", bls_scheme_str);
```
Comment (PastaPastaPasta) [inline src/rpc/evo.cpp:1341]: nit, I would prefer this to be inline, no real need for the variable

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**